src/cast_core/src/CollectionScanner.cpp

```diff
@@ -33,33 +33,61 @@
 
 namespace genny::actor {
 enum ScanType { Count, Snapshot, Standard };
+enum SortOrderType { SortNone, SortForward, SortReverse };
+using SteadyClock = std::chrono::steady_clock;
+
+// Return a vector of database names from a comma separated string
+static std::vector<std::string>
+databaseList(mongocxx::pool::entry &client, const std::string str)
+{
+    std::vector<std::string> result;
+    std::stringstream ss(str);
+    std::string token;
+
+    while (std::getline(ss, token, ',')) {
+        // Remove trailing spaces
+        token.erase(std::remove(token.begin(), token.end(), ' '), token.end());
+        result.push_back(token);
+    }
+    return result;
+}
 
 struct CollectionScanner::PhaseConfig {
-    std::vector<mongocxx::collection> collections;
+    std::vector<std::string> collectionNames;
     std::optional<DocumentGenerator> filterExpr;
-    mongocxx::database database;
+    std::vector<mongocxx::database> databases;
     bool skipFirstLoop = false;
     metrics::Operation scanOperation;
     metrics::Operation exceptionsCaught;
     int64_t documents;
     int64_t scanSizeBytes;
     ScanType scanType;
+    SortOrderType sortOrder;
+    int64_t collectionSkip;
+    std::optional<TimeSpec> scanDuration;
     bool queryCollectionList;
     std::optional<mongocxx::options::transaction> transactionOptions;
 
     PhaseConfig(PhaseContext& context,
                 const CollectionScanner* actor,
-                const mongocxx::database& db,
+                const std::vector<std::string> dbnames,
                 int collectionCount,
                 int threads,
                 bool generateCollectionNames)
-        : database{db},
+        : databases{},
           skipFirstLoop{context["SkipFirstLoop"].maybe<bool>().value_or(false)},
           filterExpr{context["Filter"].maybe<DocumentGenerator>(context, actor->id())},
           scanOperation{context.operation("Scan", actor->id())},
           exceptionsCaught{context.operation("ExceptionsCaught", actor->id())},
           documents{context["Documents"].maybe<IntegerSpec>().value_or(0)},
-          scanSizeBytes{context["ScanSizeBytes"].maybe<IntegerSpec>().value_or(0)} {
+          scanSizeBytes{context["ScanSizeBytes"].maybe<IntegerSpec>().value_or(0)},
+          collectionSkip{context["CollectionSkip"].maybe<IntegerSpec>().value_or(0)},
+          scanDuration{context["ScanDuration"].maybe<TimeSpec>()}
+    {
+        for (std::string dbname : dbnames) {
```

Comment:
I went with `const auto& dbname : dbnames`, otherwise I get:
```
error: cannot bind rvalue reference of type ‘const std::__cxx11::basic_string<char>&&’ to lvalue of type ‘std::__cxx11::basic_string<char>’
```